File: credit/credit.py
from cs50 import get_string , get_int

def main():
    card = get_string("Number: ")
    rev = card[::-1]
    if card[0]=="4" and calculate(rev) and len(card) in [13,16]:
        print("VISA")
    elif card[:2] in ["34","37"] and calculate(rev) and len(card)==15:
        print("AMEX")
    elif int(card[:2])>=51 and int(card[:2])<=55 and calculate(rev) and len(card)==16:
        print("MASTERCARD")
    else:
        print ("INVALID")


def calculate(card):
    i=0
    s=0
    ss=0
    for c in card:
        if (i%2)==1:
            ch= int(c)*2
            if ch>9:
                ch = str(ch)
                s+= int(ch[0]) + int(ch[1])
            else:
                s+= ch
        else:
            ss+=int(c)
        i += 1

    if (s+ss) % 10 == 0:
        return True
    else:
        return False

main()

File: credit/credit.c
#include <cs50.h>
#include <stdio.h>

void check_card(long n);
int check_digit(int a);
bool check_valid(long n);

int main(void)
{
    long n = get_long("Number: ");
    bool r = check_valid(n);
    if (r == true)
        check_card(n);
    else
        printf("INVALID\n");
}

bool check_valid(long n)
{
    int s1, s2, i = 0;
    while (n != 0)
    {
        if (i % 2 == 0)
        {
            s1 = s1 + (n % 10);
        }
        else
        {
            s2 = s2 + check_digit((n % 10) * 2);
        }
        n = n / 10;
        i++;
    }
    s2 = s2 + s1;
    if (s2 % 2 == 0)
        return true;
    else
        return false;
}

int check_digit(int a)
{
    if (a > 9)
    {
        int x = a % 10;
        a = x / 10;
        return x + a;
    }
    else
        return a;
}

void check_card(long n)
{
    int a = n / 10000000000000;
    int v = n / 1000000000000000;
    int vs = n / 1000000000000;
    int m = n / 100000000000000;

    if ((v == 4) || (vs == 4))
        printf("VISA\n");
    else if ((a == 34) || (a == 37))
        printf("AMEX\n");
    else if ((m >= 51) && (m <= 55))
        printf("MASTERCARD\n");
    else
        printf("INVALID\n");
}
